backend/src/externalAPI/externalAPI.service.ts

```diff
@@ -2,41 +2,74 @@ import { HttpService } from '@nestjs/axios';
 import { Injectable } from '@nestjs/common';
 import { firstValueFrom } from 'rxjs';
 import { UserContentService } from '../database/user-content/user-content.service';
-import { TypeOfAI } from '@prisma/client';
+import { TypeOfAI, User } from '@prisma/client';
+import { UserService } from '../database/user/user.service';
 
 @Injectable()
 export class ExternalAPIService {
   constructor(
     private readonly httpService: HttpService,
-    private readonly databaseService: UserContentService,
-  ) {}
+    private readonly databaseContentService: UserContentService,
+    private readonly databaseUserService: UserService,
+  ) { }
 
+  /**
+   * This function should check if a user is allowed to use the AI answer generation.
+   * @param userID
+   * @throws user ID not found | You have reached the limit for free AI responses
+   */
+  private async checkAllowed(userID: string) {
+    let user: User | null = await this.databaseUserService.getUser(userID);
+    if (user == null) {
+      throw new Error('User ID not found');
+    } else if (user.isPro == false && await this.databaseContentService.countAIAnswersForUser(userID) >= 15) {
```

Comment:
add 15 as AI_LIMIT to the config.ts in /backend

Reply:
With the AI_LIMIT variable in the config.ts, this is resolved.